/*
 * This file was automatically generated by EvoSuite
 * Mon Apr 01 04:34:16 GMT 2024
 */

package correct_java_programs;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import correct_java_programs.KNAPSACK;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class EvosuiteKNAPSACK_ESTest extends EvosuiteKNAPSACK_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      int[][] intArray0 = new int[7][1];
      int[] intArray1 = new int[5];
      intArray1[0] = 1331;
      intArray0[0] = intArray1;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(1331, intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1
         //
         verifyException("correct_java_programs.KNAPSACK", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      int[][] intArray0 = new int[3][9];
      int[] intArray1 = new int[5];
      intArray1[1] = 2;
      intArray0[0] = intArray1;
      int int0 = KNAPSACK.knapsack(2, intArray0);
      assertEquals(2, int0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      int[][] intArray0 = new int[8][5];
      // Undeclared exception!
      KNAPSACK.knapsack(1245, intArray0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(1516, (int[][]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("correct_java_programs.KNAPSACK", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      int[][] intArray0 = new int[1][5];
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack((-2619), intArray0);
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("correct_java_programs.KNAPSACK", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      int[][] intArray0 = new int[3][9];
      int[] intArray1 = new int[5];
      intArray1[0] = 2;
      intArray0[0] = intArray1;
      int int0 = KNAPSACK.knapsack(2, intArray0);
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
  }
}
